cmake/sysbuild/partition_manager.cmake

```
@@ -0,0 +1,821 @@
+      list(REMOVE_ITEM PM_${MERGED}_SPAN ${container})
+      continue()
+    endif()
+
+    # If overlapping is enabled, add the appropriate argument.
+    if(${${container}_overlap})
+      set(${container}overlap_arg --overlap=replace)
+    endif()
+
+    # Should all files still be merged ?
+    # And under which circumstances.
+    # Add command to merge files.
+    add_custom_command(
+      OUTPUT ${PROJECT_BINARY_DIR}/${container}.hex
+      COMMAND
+      ${PYTHON_EXECUTABLE}
+      ${ZEPHYR_BASE}/scripts/build/mergehex.py
+      -o ${PROJECT_BINARY_DIR}/${container}.hex
+      ${${container}overlap_arg}
+      ${${container}hex_files}
+      DEPENDS
+      ${${container}targets}
+      ${${container}hex_files}
+      # SES will load symbols from all elf files listed as dependencies to
+      # ${PROJECT_BINARY_DIR}/merged.hex. Therefore we add
+      # ${${container}elf_files} as dependency to ensure they are loaded by SES
+      # even though it is unnecessary for building the application.
+      ${${container}elf_files}
+      )
+
+    # Wrapper target for the merge command.
+    add_custom_target(
+      ${container}_hex
+      ALL DEPENDS
+      ${PROJECT_BINARY_DIR}/${container}.hex
+      )
+
+    if (DEFINED PM_DOMAIN)
+      update_runner(IMAGE ${DOMAIN_APP_${PM_DOMAIN}} HEX ${PROJECT_BINARY_DIR}/${container}.hex)
+    endif()
+
+    if ("${container}" STREQUAL "merged")
+      update_runner(IMAGE ${DEFAULT_IMAGE} HEX ${PROJECT_BINARY_DIR}/${container}.hex)
+    endif()
+  endforeach()
+
+endfunction()
+
+# Function to update an already generated runners file for a Zephyr image.
+# This function allows the build system to adjust files to flash in the config
+# section of the runners.yaml on a given image.
+function(update_runner)
+  cmake_parse_arguments(RUNNER "" "IMAGE;HEX;BIN;ELF" "" ${ARGN})
+
+  check_arguments_required("update_runner" RUNNER IMAGE)
+
+  get_target_property(bin_dir ${RUNNER_IMAGE} _EP_BINARY_DIR)
+  set(runners_file ${bin_dir}/zephyr/runners.yaml)
+
+  set(runners_content_update)
+  file(STRINGS ${runners_file} runners_content)
+  foreach(line ${runners_content})
+    if(DEFINED RUNNER_ELF AND ${line} MATCHES "^.*elf_file: .*$")
+      string(REGEX REPLACE "(.*elf_file:) .*" "\\1 ${RUNNER_ELF}" line ${line})
+    endif()
+
+    if(DEFINED RUNNER_HEX AND ${line} MATCHES "^.*hex_file: .*$")
+      string(REGEX REPLACE "(.*hex_file:) .*" "\\1 ${RUNNER_HEX}" line ${line})
+    endif()
+
+    if(DEFINED RUNNER_BIN AND ${line} MATCHES "^.*bin_file: .*$")
+      string(REGEX REPLACE "(.*bin_file:) .*" "\\1 ${RUNNER_BIN}" line ${line})
+    endif()
+    list(APPEND runners_content_update "${line}\n")
+  endforeach()
+  file(WRITE ${runners_file} ${runners_content_update})
+endfunction()
+
+
+# APP is a special domain which is handled differently.
+# Remove it from the list.
+list(REMOVE_ITEM PM_DOMAINS APP)
+
+set(user_def_pm_static ${PM_STATIC_YML_FILE})
+
+ncs_file(CONF_FILES ${APPLICATION_CONFIG_DIR}
```

`APPLICATION_CONFIG_DIR` is not defined in sysbuild. Do you want to fix it upstream? Or use:
```suggestion
ncs_file(CONF_FILES ${APP_DIR}
```

---

Maybe scratch that. In relation to https://github.com/nrfconnect/sdk-nrf/pull/11015#discussion_r1231876482, the correct change might be to use an individual image's `APPLICATION_CONFIG_DIR` (if possible) when doing per-image search for `pm_static.yml`.

---

yes, and code has been updated.
APP_DIR was just the initial adjustment to be able to continue with the static partitions handling.